x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/accesscontrol/FieldExtractorTests.java

```diff
@@ -127,34 +126,28 @@ public void testMatchNoDocs() {
         assertEquals(Collections.emptySet(), fields);
     }
 
-    public void testUnsupported() {
-        Set<String> fields = new HashSet<>();
-        expectThrows(
-            UnsupportedOperationException.class,
-            () -> { FieldExtractor.extractFields(new AssertingQuery(random(), new MatchAllDocsQuery()), fields); }
-        );
-    }
-
     public void testIndexOrDocValuesQuery() {
```

Comment:
I wanted to check the behaviour changes. We simply support all queries now, I don't see downsides. I am considering removing this test as a whole as it effectively tests that the Lucene query visitor API works, which we should probably take for granted. it may also make sense folding the utility method in the caller now that it is so small and simple.